src/main.py

```diff
@@ -1,46 +1,43 @@
-from typing import Final, Set
+from typing import Final
 from pathlib import Path
 
+import click
 import os
 
 from src.bazelize_deps import bazelize_deps
-from src.package import PackageMetadata
-from src.get_package_version import get_package_version
+from src.read_input_file import read_input_file
 
 BAZEL_WORKSPACE_DIR: Final = os.environ["BUILD_WORKSPACE_DIRECTORY"]
-DEB_PACKAGE_IN: Final = Path().joinpath(
-    BAZEL_WORKSPACE_DIR, "src", "deb_packages.in"
-)
-MODULES_DIR_IN_REGISTRY : Final = Path().joinpath(BAZEL_WORKSPACE_DIR, "modules")
 
-
-def _get_package_metadata(pinned_package: str) -> PackageMetadata:
-    if pinned_package.count(":") != 1 or pinned_package.count("=") > 1:
-        raise ValueError("Entry has unexpected format, expected format: name:arch=version or name:arch")
-    
-    # entries format: name:arch=version
-    name, arch_version = pinned_package.split(":")
-    version = ""
-    arch = arch_version
-
-    if "=" in arch_version:
-        arch, version = arch_version.split("=")
-    
-    if not version:
-        version = get_package_version(name=name, arch=arch)
-
-    return PackageMetadata(name=name, arch=arch, version=version)
-
-def _get_input_package_metadatas() -> Set[PackageMetadata]:
-    with open(DEB_PACKAGE_IN, "r") as input_file:
-        entries = set(input_file.read().splitlines())
-    
-    return {_get_package_metadata(entry) for entry in entries if entry and not entry.startswith("#")}
-   
-
-def main():
-    MODULES_DIR_IN_REGISTRY.mkdir(exist_ok=True)
-    bazelize_deps(_get_input_package_metadatas())
+@click.command(context_settings=dict(ignore_unknown_options=True))
+@click.option("--modules_path",
+               "-mp", 
+               type=click.Path(path_type=Path, file_okay=False), 
+               required=False,
+               default=Path().joinpath(BAZEL_WORKSPACE_DIR, "modules"),
+               help="The path to the directory where the local modules will reside"
+            )
+@click.option("--registry_path",
+               "-rp", 
+               type=click.Path(path_type=Path, file_okay=False), 
+               required=False,
+               default=Path().joinpath(BAZEL_WORKSPACE_DIR, "registry"),
+               help="The path to the directory where the local registry will reside"
+            )
+@click.option("--input_file",
+               "-ip", 
```

Comment:
Probably `-if` for consistency? But I'm not a big fan of short flags in general, I think only long-form args would be ok too.

Reply:
good catch ! it was input_path before 😆